.github/workflows/windows.yml

```diff
@@ -413,66 +413,134 @@ jobs:
           cmake --build bld --config '${{ matrix.type }}' --target test-ci
 
   msvc:
-    name: 'msvc (${{ matrix.arch }}, ${{ matrix.plat }}, ${{ matrix.config }})'
+    name: 'msvc vcpkg (${{ matrix.name }})'
     runs-on: windows-latest
     timeout-minutes: 30
     strategy:
       matrix:
         include:
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=ON  -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=ON ', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DBUILD_SHARED_LIBS=OFF -DENABLE_UNICODE=OFF', type: 'Debug' }
-          - { arch: 'x64', plat: 'windows', tflags: '~1516 ~2301 ~2302 ~2303 ~2307', config: '-DENABLE_DEBUG=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON          -DENABLE_UNICODE=OFF', type: 'Debug' }
+        # name should be unique and without specail chars. It create a folder with that name for cache the libreries.
+        - name: schannel
+          install: nghttp2 zstd libssh2[openssl]
+          arch: x64
+          plat: windows
+          type: Debug          
+          tflags: ~1516 ~2301 ~2302 ~2303 ~2307
+          config: -DBUILD_SHARED_LIBS=OFF -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DCURL_USE_SCHANNEL=ON -DENABLE_UNICODE=ON -DCURL_USE_LIBSSH2=ON
+        - name: openssl
+          install: nghttp2 nghttp3 zstd libssh2[openssl]
+          arch: x64
+          plat: windows
+          type: Debug          
+          tflags: ~1516 ~2301 ~2302 ~2303 ~2307
+          config: -DBUILD_SHARED_LIBS=OFF -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DENABLE_UNICODE=ON -DCURL_USE_LIBSSH2=ON -DCURL_USE_OPENSSL=ON -DUSE_OPENSSL_QUIC=ON
+        - name: no ssl
+          install: nghttp2 zstd zlib
+          arch: x64 
+          plat: windows
+          type: Debug
+          tflags: ~1516 ~2301 ~2302 ~2303 ~2307
+          config: -DENABLE_DEBUG=ON -DCURL_ZSTD=ON -DUSE_NGHTTP2=ON -DCURL_USE_SCHANNEL=OFF -DHTTP_ONLY=ON  -DENABLE_UNICODE=OFF
```

Comment:
I suggest using single quotes for all the above strings. It helps syntax highlighters and linters.